config/config.go

```diff
@@ -23,14 +22,24 @@ import (
 	"strings"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws/ec2metadata"
-	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/prometheus/prometheus/discovery/azure"
+	"github.com/prometheus/prometheus/discovery/consul"
+	"github.com/prometheus/prometheus/discovery/dns"
+	"github.com/prometheus/prometheus/discovery/ec2"
+	"github.com/prometheus/prometheus/discovery/file"
+	"github.com/prometheus/prometheus/discovery/gce"
+	"github.com/prometheus/prometheus/discovery/marathon"
+	"github.com/prometheus/prometheus/discovery/triton"
+	"github.com/prometheus/prometheus/discovery/zookeeper"
+
 	"github.com/prometheus/common/model"
-	"gopkg.in/yaml.v2"
+	"github.com/prometheus/prometheus/pkg/targetgroup"
+	configUtil "github.com/prometheus/prometheus/util/config"
```

Comment:
This code is needed across the project, I believe it should go in the common repository.

Reply:
I think I agree... but IIRC @fabxc dislikes putting too much into common and prefers duplication? Would be good to get his opinion too.

Reply:
This library includes security-sensitive code, that's not something we should be duplicating. Duplication also leads to drift.

Reply:
Makes sense to me. But how about still keeping the code in this repo for this PR and doing the splitout separately. Keeps this PR simpler.

Reply:
That makes sense. This PR is going to be intricate enough as-is.